components/match2/wikis/clashroyale/match_group_input_custom.lua

```diff
@@ -0,0 +1,724 @@
+local ALLOWED_STATUSES = {'W', 'FF', 'DQ', 'L'}
+local CONVERT_STATUS_INPUT = {W = 'W', FF = 'FF', L = 'L', DQ = 'DQ', ['-'] = 'L'}
+local DEFAULT_LOSS_STATUSES = {'FF', 'L', 'DQ'}
+local MAX_NUM_OPPONENTS = 2
+local MAX_NUM_PLAYERS_PER_MAP = 2
+local DEFAULT_BEST_OF = 99
+local EPOCH_TIME_EXTENDED = '1970-01-01T00:00:00+00:00'
+local TBD = 'tbd'
+local TBD_PLAYERS = {TBD, 'tba'}
+
+local CustomMatchGroupInput = {}
+
+-- called from Module:MatchGroup
+function CustomMatchGroupInput.processMatch(match)
+	Table.mergeInto(
+		match,
+		CustomMatchGroupInput._readDate(match)
+	)
+	match = CustomMatchGroupInput._getExtraData(match)
+	match = CustomMatchGroupInput._getTournamentVars(match)
+	match = CustomMatchGroupInput._checkFinished(match)
+	match = CustomMatchGroupInput._adjustData(match)
+	match = CustomMatchGroupInput._getVodStuff(match)
+	match = CustomMatchGroupInput._getLinks(match)
+
+	return match
+end
+
+function CustomMatchGroupInput._readDate(matchArgs)
+	if matchArgs.date then
+		return MatchGroupInput.readDate(matchArgs.date)
+	else
+		return {
+			date = EPOCH_TIME_EXTENDED,
+			dateexact = false,
+			timestamp = DateExt.epochZero,
+		}
+	end
+end
+
+function CustomMatchGroupInput._checkFinished(match)
+	if Logic.readBoolOrNil(match.finished) == false then
+		match.finished = false
+	elseif Logic.readBool(match.finished) then
+		match.finished = true
+	elseif Logic.isNotEmpty(match.winner) then
+		match.finished = true
+	end
+
+	-- Match is automatically marked finished upon page edit after a
+	-- certain amount of time (depending on whether the date is exact)
+	if match.finished ~= true then
+		local currentUnixTime = os.time(os.date('!*t'))
+		local threshold = match.dateexact and 30800 or 86400
+		if match.timestamp + threshold < currentUnixTime then
+			match.finished = true
+		end
+	end
+
+	return match
+end
+
+function CustomMatchGroupInput._getTournamentVars(match)
+	match.mode = Logic.emptyOr(match.mode, Variables.varDefault('tournament_mode', 'solo'))
+	match.publishertier = Logic.emptyOr(match.publishertier, Variables.varDefault('tournament_publishertier'))
+	return MatchGroupInput.getCommonTournamentVars(match)
+end
+
+function CustomMatchGroupInput._getVodStuff(match)
+	match.stream = Streams.processStreams(match)
+	match.vod = Logic.emptyOr(match.vod)
+
+	return match
+end
+
+function CustomMatchGroupInput._getLinks(match)
+	match.links = {
+		preview = match.preview,
+		preview2 = match.preview2,
+		interview = match.interview,
+		interview2 = match.interview2,
+		review = match.review,
+		recap = match.recap,
+		lrthread = match.lrthread,
+	}
+	return match
+end
+
+function CustomMatchGroupInput._getExtraData(match)
+	local extradata = {
+		casters = match.casters,
+		t1bans = CustomMatchGroupInput._readBans(match.t1bans),
+		t2bans = CustomMatchGroupInput._readBans(match.t2bans),
+		hasbigmatch = Logic.readBool(match.hasBigMatch) or nil,
+	}
+
+	for subGroupIndex = 1, MAX_NUM_MAPS do
```

Comment:
max num of subgroupings should be `<` than max num maps right?

Reply:
technically no
realistically yes but the max number of that is not really known as it depends on the format